ksql-rest-app/src/main/java/io/confluent/ksql/rest/util/ClusterTerminator.java

```diff
@@ -0,0 +1,161 @@
+      final List<String> sourcesList,
+      final boolean isKeep
+  ) {
+    // Terminate all queries
+    getListForSet(ksqlEngine).forEach(
+        queryMetadata -> {
+          if (queryMetadata instanceof PersistentQueryMetadata) {
+            final PersistentQueryMetadata persistentQueryMetadata
+                = (PersistentQueryMetadata) queryMetadata;
+            ksqlEngine.terminateQuery(persistentQueryMetadata.getQueryId(), true);
+          }  else {
+            queryMetadata.close();
+          }
+        }
+    );
+
+    // if we have the explicit list of stream/table to delete.
+    final MetaStore metaStore = ksqlEngine.getMetaStore();
+    if (sourcesList.isEmpty()) {
+      metaStore.getAllStructuredDataSources().forEach((s, structuredDataSource) ->
+          deleteSource(s, structuredDataSource, ksqlEngine));
+    } else {
+      if (isKeep) {
+        metaStore.getAllStructuredDataSources().forEach(
+            (sourceName, structuredDataSource) -> {
+              if (!sourcesList.contains(sourceName)) {
+                deleteSource(sourceName, structuredDataSource, ksqlEngine);
+              }
+            }
+        );
+      } else {
+        sourcesList
+            .forEach(sourceName -> deleteSource(
+                sourceName,
+                metaStore.getSource(sourceName),
+                ksqlEngine));
+      }
+    }
+  }
+
+  // This is needed because the checkstyle complains if we create this in place.
+  private List<QueryMetadata> getListForSet(
+      final KsqlEngine ksqlEngine
+  ) {
+    final List<QueryMetadata> queryMetadataList = new ArrayList<>();
+    queryMetadataList.addAll(ksqlEngine.getAllLiveQueries());
+    return queryMetadataList;
+  }
+
+  private void deleteSource(
+      final String sourceName,
+      final StructuredDataSource structuredDataSource,
+      final KsqlEngine ksqlEngine) {
+    final DdlCommand ddlCommand = new DropSourceCommand(
+        getAbstractStreamDropStatement(sourceName,
+            structuredDataSource.getDataSourceType() == DataSourceType.KSTREAM),
+        structuredDataSource.getDataSourceType(),
+        ksqlEngine.getTopicClient(),
+        ksqlEngine.getSchemaRegistryClient(),
+        true
+    );
+    final DdlCommandExec ddlCommandExec = ksqlEngine.getDdlCommandExec();
+    ddlCommandExec.execute(ddlCommand, false);
+  }
+
+  private AbstractStreamDropStatement getAbstractStreamDropStatement(
+      final String sourceName,
+      final boolean isStream) {
+    if (isStream) {
+      return new DropStream(QualifiedName.of(sourceName), false, true);
+    } else {
+      return new DropTable(QualifiedName.of(sourceName), false, true);
+    }
+  }
+
+  private void deleteCommandTopic(
+      final KsqlConfig ksqlConfig,
+      final KsqlEngine ksqlEngine,
+      final Consumer<CommandId, Command> commandConsumer,
+      final Producer<CommandId, Command> commandProducer) {
+    // Delete the command topic
+    commandConsumer.close();
```

Comment:
why are we passing these two all the way down to here just to close them? Can we not just close them in `StatementExecutor`??

Reply:
Yes, we can. Now closing them in `StatementExecutor`.